WooCommerce/Resources/Info.plist

```diff
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>${VERSION_SHORT}</string>
+	<string>$(MARKETING_VERSION)</string>
```

Comment:
This change, and the one below to use `CURRENT_PROJECT_VERSION` instead of `VERSION_LONG` were not strictly necessary for the versions to be in sync.

Still, by using the standard Xcode values here, and by re-mapping our `VERSION_SHORT` and `VERSION_LONG` to them, we should be able to keep the additional setup to a minimum. That is shown by there not being any extra settings to configure the version for the widget target below.

Reply:
I'm not sure I follow how using `MARKETING_VERSION` instead of `VERSION _SHORT` then having to map `MARKETING_VERSION=$VERSION_SHORT` in `.xcconfig` is better than using `VERSION_SHORT` directly, especially since we'd get those values directly from the xcconfig if we don't have those defined at neither target nor project level (like we don't for `MARKETING_VERSION` after your deletion of those at that level)?

Reply:
I think there's two questions here: Why use `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` at all, and why use `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` in the WooCommerce target.

### Why use `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` at all? 

As far as I understand, `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` are the default build settings  (as of Xcode 11?) to use for versioning. 

When creating a new target, Xcode 13 puts these version settings in the project file and doesn't add `CFBundleVersion` and `CFBundleShortVersionString` to the `Info.plist`.

https://github.com/woocommerce/woocommerce-ios/blob/1f424f02de88edde4791d2156d655ff2bda096bc/WooCommerce/StoreWidgets/Info.plist#L1-L11

By relying on those two settings, we can control the version in a centralized way across any number of future targets.

Without the re-mapping, we'd have to do something like `CURRENT_PROJECT_VERSION = $(VERSION_LONG)` for every new target for every build configuration.

Configuring the version through those values directly seems more economic.

### Why use `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` in the WooCommerce target?

Consistency. If we agree that setting `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` only once in one `xcconfig` is the way to go, then it makes sense to have them across all targets.

`VERSION_SHORT` and `VERSION_LONG` have to stay because that's what the release toolkit uses, for the time being at least, but are implementation details that don't need to leak in the rest of the project.

Reply:
Ah, gotcha.

I didn't notice that you used `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` in the WooCommerce target as well (and not just in the new Widget target), so I thought you actually introduced an inconsistency (`MARKETING_VERSION` in new widget target but `VERSION_SHORT` in current `WooCommerce` target) while if fact you did the opposite and actually _resolved_ the inconsistency, right?

So do I understand correctly that `VERSION_SHORT` and `VERSION_LONG` are not used anywhere in the `pbxproj` file? So their _only_ reference is in the `.xcconfig` file (for the release toolkit to find them), but their only usage is also in the same `.xcconfig` file when you do `MARKETING_VERSION=VERSION_SHORT`… and then `MARKETING_VERSION` is what's used in the `pbxproj` at all levels (i.e. in project and all targets), never `VERSION_SHORT` there?

If you confirm that's the case then I'm all 👍  for that approach then!

_(And in a far future we might imagine the various actions that bump or read the versions to be allowed to take a parameter telling the name of the build setting to use so we would be able to drop the mapping in `.xcconfig` and make the release-toolkit directly bump `MARKETING_VERSION` in the `.xcconfig`… right?)_

Reply:
> So do I understand correctly that `VERSION_SHORT` and `VERSION_LONG` are not used anywhere in the `pbxproj` file?

That is correct:

![image](https://user-images.githubusercontent.com/1218433/188758322-331e0dc8-2e73-4961-b6ce-125724188fbe.png)

> _(And in a far future we might imagine the various actions that bump or read the versions to be allowed to take a parameter telling the name of the build setting to use so we would be able to drop the mapping in `.xcconfig` and make the release-toolkit directly bump `MARKETING_VERSION` in the `.xcconfig`… right?)_

Yes 🤩